Samples/HelloWorld/OrleansHost/OrleansHostWrapper.cs

```diff
@@ -0,0 +1,171 @@
+        }
+
+        public int Stop()
+        {
+            if (siloHost != null)
+            {
+                try
+                {
+                    siloHost.StopOrleansSilo();
+                    siloHost.Dispose();
+                    Console.WriteLine("Orleans silo '{0}' shutdown.", siloHost.Name);
+                }
+                catch (Exception exc)
+                {
+                    siloHost.ReportStartupError(exc);
+                    Console.Error.WriteLine("{0}:\n{1}\n{2}", exc.GetType().FullName, exc.Message, exc.StackTrace);
+                    return 1;
+                }
+            }
+            return 0;
+        }
+
+        private SiloArgs ParseArguments(string[] args)
+        {
+            string deploymentId = null;
+
+            string siloName = null;
+
+            for (int i = 0; i < args.Length; i++)
+            {
+                string arg = args[i];
+                if (arg.StartsWith("-") || arg.StartsWith("/"))
+                {
+                    switch (arg.ToLowerInvariant())
+                    {
+                        case "/?":
+                        case "/help":
+                        case "-?":
+                        case "-help":
+                            // Query usage help
+                            PrintUsage();
+                            return null;
+                        default:
+                            Console.WriteLine("Bad command line arguments supplied: " + arg);
+                            return null;
+                    }
+                }
+                else if (arg.Contains("="))
+                {
+                    string[] parameters = arg.Split('=');
+                    if (String.IsNullOrEmpty(parameters[1]))
+                    {
+                        Console.WriteLine("Bad command line arguments supplied: " + arg);
+                        return null;
+                    }
+                    switch (parameters[0].ToLowerInvariant())
+                    {
+                        case "deploymentid":
+                            deploymentId = parameters[1];
+                            break;
+                        case "name":
+                            siloName = parameters[1];
+                            break;
+                        default:
+                            Console.WriteLine("Bad command line arguments supplied: " + arg);
+                            return null;
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Bad command line arguments supplied: " + arg);
+                    return null;
+                }
+            }
+
+            return new SiloArgs(siloName ?? Dns.GetHostName(), deploymentId); // Default to machine name
+        }
+
+        public void PrintUsage()
+        {
+            Console.WriteLine(
+                @"USAGE: 
+    [OrleansHost] [name=<siloName>] [deploymentId=<idString>] [/debug]
```

Comment:
concatenate `Assembly.GetExecutingAssembly().Name` instead of hard-coding OrleansHost, since the console app name might be different.
Also, the exe name is not optional, so remove the brackets.